test/test_parser.cpp

```diff
@@ -781,6 +780,9 @@ TEST(Parser_LinksToDifferentTable)
     list_2.add(item_keys[2]);
     list_2.add(item_keys[3]);
 
+    verify_query(test_context, t, "items = obj('Items', 'coffee')", 0); // nobody buys coffee
+    verify_query(test_context, t, "items = obj('Items', 'milk')", 2);   // but milk
+    verify_query(test_context, t, "items = O0", 2);                     // how many people bought milk?
```

Comment:
i think we should also have a couple tests that use other primary key types instead of just string (number, objectID, uuid)

Reply:
We have that elsewhere

Reply:
I know we have tests that parse a primary key of those types in RQL, but as far as I can tell we don't have any that use those from within the new `obj( ... )` expression. Would that not be useful? Or, if they do exist somewhere else, maybe you could point them out to me, I'm not as familiar with this codebase.

Reply:
@mpobrien there are a few here: https://github.com/realm/realm-core/pull/5835/files#diff-eed0db0ed99a211ac7d34de95f2a953e1e3ba234b419ce1726bc629e00e740cbR5552-R5600
Is that sufficient?

Reply:
Ah OK, that's what I was missing. Thanks!